src/core/public/application/ui/app_container.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { Subject } from 'rxjs';
+
+import { LegacyApp, AppMounter, AppUnmount } from '../types';
+import { HttpStart } from '../../http';
+import { AppNotFound } from './app_not_found_screen';
+
+interface Props extends RouteComponentProps<{ appId: string }> {
+  apps: ReadonlyMap<string, AppMounter>;
+  legacyApps: ReadonlyMap<string, LegacyApp>;
+  basePath: HttpStart['basePath'];
+  currentAppId$: Subject<string | undefined>;
+  /**
+   * Only necessary for redirecting to legacy apps
+   * @deprecated
+   */
+  redirectTo: (path: string) => void;
+}
+
+interface State {
+  appNotFound: boolean;
+}
+
+export class AppContainer extends React.Component<Props, State> {
+  private readonly containerDiv = React.createRef<HTMLDivElement>();
+  private unmountFunc?: AppUnmount;
+
+  constructor(props: Props) {
```

Comment:
I think this can be shortened to a class field:

```diff
- constructor(props: Props) {
-   super(props);
-   this.state = { appNotFound: false };
- }
+ state = { appNotFound: false };
```